### ISSUE TITLE
On Orientation Change Issue

### ISSUE THREAD

**Issue by mh6300** [jfeinstein10/SlidingMenu#11]
Hi, 

First off thanks for the library, I plan on using it in my app EzImgur - I am stuck on this one issue and was hoping you could maybe point me in the right direction (as far as troubleshooting and fixing). 

Long story short - I am using the menu on an activity that does not restart on orientation change (android:configChanges="keyboardHidden|orientation|screenSize"). I am using the sliding menu view component (I am not using any of the custom activity classes) as the root view component in my activities and I am setting the "above" content by setting the view above each activity in the set content view.

As far as the issue: 
Normally when I hit the home icon on the action bar, it will open the viewBehind and hide the action bar. This all works fine until the screen orientation changes. When the menu is not open and the screen orientation changes - the menu content is magically visible. When I check the sliding menu object to see if the behind view is visible I get false. If i try to hide the menu it does not work (probably because the menu thinks it is not open). 

To sum up:  

The behind view is being shown on orientation change (even when it was not open before the screen orientation change), and the sliding menu does not even think it is showing - I am unable to hide the menu unless the user clicks a menu item. 

Screenshots:
http://imgur.com/a/MsThb
Pic 1: the app normally before opening menu. 
Pic 2: the app after opening menu. 
Pic 3: the app after orientation change (the menu was not open when I changed orientation.)

**Comment by mh6300** [comment]
Just to note: Everything works fine on other activities that use the normal lifecycle. I only see the issue on the main gallery activity, with the special orientation config. 

The logic for using the menu is the same for all activities and is abstracted to a super class that all of the activities extend. Source code for super activity: https://gist.github.com/3119992

Activity Config:
        <activity
            android:name=".activity.GalleryActivity"
            android:configChanges="keyboardHidden|orientation|screenSize"
            android:label="@string/app_name"
            android:launchMode="singleTask">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>

**Comment by jfeinstein10** [comment]
So this only happens in your GalleryActivity which extends the super class Activity you sent me. And only happens when you maintain the Activity's configuration on orientation change?
If you don't maintain the configuration on orientation change, it works correctly?

**Comment by mh6300** [comment]
That is correct, the gallery activity is the only one where I am handling the configuration change - and is the only one that is having the issue.

**Comment by jfeinstein10** [comment]
Probably has something to do with saving the menu state. I don't know how quickly I'll be able to get to it during the work week.

**Comment by mh6300** [comment]
Thanks and no worries, I will try and dig in to see if I can figure it out in the meantime.

**Comment by jfeinstein10** [comment]
I would write something to save the state of the SlidingMenu as a whole. I'll end up just getting rid of the crappy stuff that's lower down in the CustomViews

**Comment by mh6300** [comment]
Just to update - I tried some things, but never got anywhere (wasn't sure what state to save really as I am fairly new to Android and was not sure what state would be applicable).

To get the app update out, I ended up just switching the gallery activity to not use the "special"/not suggested way of handling orientation change. Now I am instead just saving the state of my gallery activity (e.g. the data I loaded over the network) via retain non custom configuration instance, which helps get rid of the issue for me all together. 

However, I would like to know what you had in mind for saving state. Maybe I can help fix the issue. 

Thanks again for the work you have done.

**Comment by jfeinstein10** [comment]
Pull the changes that I have in there now and let me know if it solves this issue. I just wrote very simple state saving into the SlidingMenu class. Basically it only saves whether or not the menu was open.

Also, what app is this? I'd love to feature some apps that are using the library.

**Comment by mh6300** [comment]
Awesome thanks, I will update if it fixes it later tonight. App is called EZ Imgur. Basically a simple imgur.com browser (https://play.google.com/store/apps/details?id=com.ezimgur&hl=en).

**Comment by jfeinstein10** [comment]
Cool. Just downloaded to check it out.
Also, hate to be the police on this one but could you please credit both SlidingMenu and ActionBarSherlock and include their licenses in your next update. Here's a good example license page (from SongKick) : http://i.imgur.com/PioXY.png

**Comment by mh6300** [comment]
Sorry about that, I wasn't entirely sure what was needed so I credited both in the app store page. Will add to app itself tonight.

**Comment by mh6300** [comment]
Fix did not work when I used the orientation/screen size config and added the state saving code (I did not do a full check out -- so maybe I was missing some changes - I will get latest tomorrow and try). Also - I just pushed a build with the license information in the app (help screen), should be in the market soon.

**Comment by chrisjenx** [comment]
This has got me too, Can't seem to work out what the issue is, I have debugging on and can see it layout and have the correct pages set.

**Comment by ChrisM001** [comment]
After changing orientation I get a kind of doubled view behind.

https://www.dropbox.com/s/ckww47qhrclcz5l/Screenshot_2012-07-29-10-52-48.png

**Comment by jfeinstein10** [comment]
That's just because I did some really crappy Fragment management. And by crappy I mean none haha. Doesn't have anything to do with the SlidingMenu. I'll need to fix that though.

**Comment by chrisjenx** [comment]
Ha the issue still stands as it gets stuck open. I'm going to have a good
look today
 On 29 Jul 2012 10:29, "Jeremy Feinstein" <
reply@reply.github.com>
wrote:

**Comment by chrisjenx** [comment]
Fixed this in pull https://github.com/jfeinstein10/SlidingMenu/pull/39 if people could test that would be grand!

**Comment by cdzombak** [comment]
Should this be closed now that it appears #39 has been accepted?

**Comment by chrisjenx** [comment]
Yeah, go for it, people seem to be using it without any issues.
